app/lib/pdf/printer.rb

```diff
@@ -12,16 +12,23 @@ module Pdf::Printer
   # Creates 2-columns layout. Calls the block with :left or :right
   # parameter to render the columns content.
   #
-  def two_columns(at = [0.mm, @pdf.cursor], options = {}, &block)
-    @pdf.bounding_box(at, { :width => PAGE_WIDTH }.merge(options)) do
-      @pdf.column_box [0.mm, @pdf.bounds.top], :width => TABLE_HALF_WIDTH  do
+  def two_columns(at = [0.mm, @pdf.cursor], options = {})
+    first_y = 0
+    second_y = 0
+
+    box = @pdf.bounding_box(at, { :width => PAGE_WIDTH }.merge(options)) do
+      @pdf.column_box [0.mm, @pdf.bounds.top], :width => TABLE_HALF_WIDTH, columns: 1 do
         yield(:left)
       end
+      first_y = @pdf.y
 
-      @pdf.column_box [97.mm, @pdf.bounds.top], :width => TABLE_HALF_WIDTH do
+      @pdf.column_box [TABLE_HALF_WIDTH + 7.mm, @pdf.bounds.top], :width => TABLE_HALF_WIDTH, columns: 1 do
         yield(:right)
       end
+      second_y = @pdf.y
     end
+
+    @pdf.move_down second_y - first_y if first_y < second_y && box.stretchy?
```

Comment:
This will always print 50/50 width columns, right? Could it be like this?:

- Left column (header): adjusted to the size of the content, with a maximum of 50%
- Right column (data): remaining space, new line if the content doesn't fit.

I don't remember if there's a straightforward way to do this, I know Prawn is a PITA so if it's too complicated forget what I said.

Reply:
This is different. It is not a single table with header and values. It is 2 separate random PDF contents that should be presented next to each other. We use it presently only to present two tables next to each other. In the attached PDFs you can see how we have two independent tables next to each other.

Reply:
Ah I understand, I thought it was a table